curator-test/src/main/java/org/apache/curator/test/BaseClassForTests.java

```diff
@@ -123,7 +123,16 @@ public void setup() throws Exception
         System.setProperty(INTERNAL_PROPERTY_REMOVE_WATCHERS_IN_FOREGROUND, "true");
         System.setProperty(INTERNAL_PROPERTY_VALIDATE_NAMESPACE_WATCHER_MAP_EMPTY, "true");
 
-        createServer();
+        try
+        {
+            createServer();
+        }
+        catch ( FailedServerStartException ignore )
+        {
+            log.warn("Failed to start server - retrying");
+            closeServer();
+            createServer();
```

Comment:
Nice to document that we try exactly one more time for a best effort flaky test auto-recovery. It is confused at the first glance(why retry exactly once) and over-engineering if we set the retry time as a field or even configurable.